code/game/objects/items/devices/taperecorder.dm

```diff
@@ -40,6 +41,18 @@
 		mytape = I
 		to_chat(user, "<span class='notice'>You insert [I] into [src].</span>")
 		update_icon()
+	else if(is_wire_tool(I) && panel_open)
+		wires.interact(user)
+
+/obj/item/taperecorder/screwdriver_act(mob/living/user, obj/item/I)
+	if(I.tool_behaviour == TOOL_SCREWDRIVER)
+		I.play_tool_sound(src, 50)
+		if(!panel_open)
+			panel_open = TRUE
+			to_chat(user, "<span class='notice'>You open the maintenance hatch of [src].</span>")
+		else
+			panel_open = FALSE
+			to_chat(user, "<span class='notice'>You close the maintenance hatch of [src].</span>")
```

Comment:
```suggestion
/obj/item/taperecorder/screwdriver_act(mob/living/user, obj/item/I)
	I.play_tool_sound(src, 50)
	if(!panel_open)
		panel_open = TRUE
		to_chat(user, "<span class='notice'>You open the maintenance hatch of [src].</span>")
	else
		panel_open = FALSE
		to_chat(user, "<span class='notice'>You close the maintenance hatch of [src].</span>")
	return TRUE
```
* screwdriver_act() will only be called if the tool has this kind of behavior.
* We return a truthy value to end the attack chain.